### DISCUSSION TITLE
One document, several signatures in one API call

### DISCUSSION THREAD

**Discussion by jokamax** [Keyfactor/signserver-ce#97] [Q&A]
Hi,

In the documentation I read they can send for one call :

```
REQUEST_METADATA.VISIBLE_SIGNATURE_RECTANGLE
REQUEST_METADATA.VISIBLE_SIGNATURE_PAGE
```

For a call of type :

```
MultipartBody prepareRequest = Unirest.post(URL_API_PRODUCTION + "/process")
                .field("workerName", "PDFSigner")
                .field("processType", "signDocument")
                .field("filereceivefile", file);
```

But in some case, we have two or more signature one the same document.
Should we send an API call "for each signature"? or is there a way to send a document one time only with several "page/position"?

thanks

**Comment by netmackan** [comment]
The PDFSigner only adds one signature at a time.
You can make one call and then one again but then you would end up with a PDF with two signatures/revisions made by the same signer which seems redundant.
Maybe what you want is one digital signature but the visible representation in multiple places? That however, is not currently supported by SignServer but if someone knows if/how that can be done please let us know.

**Reply by jokamax** [reply]
In fact, we have PDF documents with multiple signature placements (accounting document for example).

Currently, we "resend" the document to be signed to the signature server for each location to be signed which is a shame. 1 document, 3 signature locations, the person signs 3 times: this sends the document to our server with the 3 signatures to be made: 3 calls necessary to the signserver

I have already used software that allowed a single round trip and I was hoping that the signserver would allow me to do that too.

**Reply by netmackan** [reply]
Sounds like a good feature. I am wondering of those 3 different visible signatures really are tied to the signature or two of them are just normal images... Maybe it is possible to have multiple "real" visible signatures. Maybe someone here knows and how to do that with OpenPDF?
